lib/elastomer/client/percolator.rb

```diff
@@ -12,6 +12,14 @@ def initialize(client, index_name, id)
         @client = client
         @index_name = client.assert_param_presence(index_name, "index name")
         @id = client.assert_param_presence(id, "id")
+
+        # COMPATIBILITY
+        @percolator_type =
+          if client.version_support.es_version_5_x?
```

Comment:
Here's how I'd recommend doing this:

1. Add a new method to `VersionSupport` called `percolator_type` with this code.
2. Call it where ever you need the percolator type.

This way we can keep the conditional logic isolated.

I thought about making `es_version_2_x?` and `es_version_5_x?` private to discourage their use outside of `VersionSupport` but I thought we might need them for something.

Reply:
that makes sense, will update!